### PR TITLE
Added changes from flyteplugins and also added functionality for FetchFromLiteral

### PR DESCRIPTION
**Pull request by pmahindrakar-oss**
# TL;DR
Following files in flyteplugins have additional functionality w.r.t coreutils which should ideally be present in flyteidl
https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/utils/literals.go
https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/utils/literals_test.go

This MR moves all the additional functionality 
Which will be used for flytectl create executions command.

Only cosmetic differences remain now between the literals.go in  flyteplugins against the one in flyteidl coreutils.
We should deprecate and remove the usage of this from flyteplugins and instead rely on flyteidl version

```
sdiff -s flyteplugins/go/tasks/pluginmachinery/utils/literals.go flyteidl/clients/go/coreutils/literals.go 
package utils						      |	// Contains convenience methods for constructing core types.
							      >	package coreutils
							      >		"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
							      >		"github.com/lyft/flytestdlib/storage"
							      >
	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"    <
	"github.com/lyft/flytestdlib/storage"		      <
	return nil, errors.Errorf("Failed to convert to a kno |		return nil, fmt.Errorf("failed to convert to a known 
							      <
			// case *core.LiteralType_Schema:     |				//case *core.LiteralType_Schema:
	return nil, errors.Errorf("Failed to convert to a kno |		return nil, fmt.Errorf("failed to convert to a known
```

Just structural import diff between the unit tests  which is for fixing linter issue in flyteidl.
```
sdiff -s flyteplugins/go/tasks/pluginmachinery/utils/literals_test.go flyteidl/clients/go/coreutils/literals_test.go 
package utils						      |	package coreutils
	"github.com/golang/protobuf/ptypes"		      <
	structpb "github.com/golang/protobuf/ptypes/struct"   <
							      >
							      >		"github.com/golang/protobuf/ptypes"
							      >		structpb "github.com/golang/protobuf/ptypes/struct"
```

Also additionally added functionality to FetchFromLiteral 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin
 - [x] Tech Debt

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/<number>

## Follow-up issue
https://github.com/flyteorg/flyte/issues/757

### CODE REVIEW

File: clients/go/coreutils/fetch_literal.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package coreutils
+
+import (
+	"fmt"
+
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+)
+
+func FetchFromLiteral(literal *core.Literal) (interface{}, error) {
```
Comment (kumare3) [inline clients/go/coreutils/fetch_literal.go:9]:
```suggestion
func ExtractFromLiteral(literal *core.Literal) (interface{}, error) {
```

File: clients/go/coreutils/fetch_literal.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package coreutils
```
Comment (kumare3) [inline clients/go/coreutils/fetch_literal.go:1]: For this package, I think we should start adding a top-level comments (I know we dont have it, but can you please help us set the precedent)
so essentially before the package
// extract_literal.go
// Utility methods to extract a native golang value from a given Literal.
// Usage:
// .....

File: clients/go/coreutils/fetch_literal.go (hunk 1 above)
Comment (pmahindrakar-oss) [reply clients/go/coreutils/fetch_literal.go]: Done

File: clients/go/coreutils/fetch_literal.go (hunk 2 above)
Comment (pmahindrakar-oss) [reply clients/go/coreutils/fetch_literal.go]: Done.

**Review by kumare3 — APPROVED** [review]
